packages/blockchain-wallet-v4-frontend/src/data/components/exchange/sagas.utils.js

```diff
@@ -116,8 +118,8 @@ export default ({ api, coreSagas }) => {
         if (isUndefinedOrEqualsToZero(sourceAmount)) return defaultResult
         const quotation = yield call(api.createQuote, sourceAmount, pair, true)
         const targetAmount = path(['success', 'withdrawalAmount'], quotation) || 0
-        const sourceFiat = convertCoinToFiat(sourceAmount, sourceCoin, sourceCoin, 'USD', sourceRates).value
-        const targetFiat = convertCoinToFiat(targetAmount, targetCoin, targetCoin, 'USD', targetRates).value
+        const sourceFiat = convertCoinToFiat(sourceAmount, sourceCoin, sourceCoin, currency, sourceRates).value
+        const targetFiat = convertCoinToFiat(targetAmount, targetCoin, targetCoin, currency, targetRates).value
         return { sourceAmount, sourceFiat, targetAmount, targetFiat }
```

Comment:
In v3 the currency is USD if the user is in the US, EUR otherwise, not sure we should change this. Also there is a limit set from wallet-options of 500 (USD if user is US, EUR otherwise)

Reply:
I never heard about the 500 limits ? does it mean we can't validate the trade if fiat amount > 500 ?

Reply:
Sorry I should have brought it up before
1. the form should be invalid if amount > 500
2. fiat = country_code === US ? USD : EUR